build.sh

```diff
@@ -58,53 +162,67 @@ else
     KSP_NAME=${ghprbActualCommit}
 fi
 
-mono --debug ckan.exe ksp add ${KSP_NAME} "`pwd`/dummy_ksp"
-mono --debug ckan.exe ksp default ${KSP_NAME}
-
-echo Running ckan update
-mono --debug ckan.exe update
-
-echo Running jsonlint on the changed files
-echo If you get an error below you should look for syntax errors in the metadata
-
-jsonlint -s -v ${COMMIT_CHANGES}
-
-echo Fetching latest netkan.exe
-
-# fetch latest netkan.exe
-wget --quiet http://ci.ksp-ckan.org:8080/job/NetKAN/lastSuccessfulBuild/artifact/netkan.exe -O netkan.exe
-
-mkdir built
-
-# additional NETKAN_OPTIONS may be set on jenkins jobs
+# Build all the passed .netkan files.
+# Note: Additional NETKAN_OPTIONS may be set on jenkins jobs
 for f in ${COMMIT_CHANGES}
 do
-	echo Running NetKAN for $f
-	mono --debug netkan.exe $f --cachedir="dummy_ksp/CKAN/downloads" --outputdir="built" ${NETKAN_OPTIONS}
+    echo "Running NetKAN for $f"
+    mono netkan.exe $f --cachedir="downloads_cache" --outputdir="built" ${NETKAN_OPTIONS}
 done
 
-for f in built/*.ckan
-do
-	echo ----------------------------------------------
-	echo 
-	cat $f | python -m json.tool
-	echo ----------------------------------------------
-	echo 
-	echo Running ckan install -c $f
-	mono --debug ckan.exe install -c $f --headless
-done
+# Reset KSP.
+mono ckan.exe ksp forget $KSP_NAME_DEFAULT
 
-
-echo ----------------------------------------------
-echo
-ADDON_LIST=$(mono ckan.exe list --porcelain)
-OFS=${IFS}
-IFS=' '
-echo ${ADDON_LIST} | while read state name version
+# Test all the built files.
+for f in built/*.ckan
 do
-    # echo "state name version '${state}' '${name}' '${version}'"
-    mono --debug ckan.exe show ${name}
-    echo ----------------------------------------------
-    echo
+    echo "Checking $f"
+    echo "----------------------------------------------"
+    echo ""
+    cat $f | python -m json.tool
+    echo "----------------------------------------------"
+    echo ""
+    
+    # Get a list of all the OTHER files.
+    OTHER_FILES=()
+    
+    for o in built/*.ckan
+    do
+        if [ "$f" != "$o" ]
+        then
+            OTHER_FILES+=($o)
+        fi
+    done
+    
+    # Inject into metadata.
+    inject_metadata $OTHER_FILES
+    
+    # Extract identifier and KSP version.
+    CURRENT_IDENTIFIER=$(./jq '.identifier' $f)
+    CURRENT_KSP_VERSION=$(./jq 'if .ksp_version then .ksp_version else .ksp_version_min end' $f)
+    
```

Comment:
The docker image will need 'jq' to be installed, but that won't be difficult. However ```./jq``` will cause it to try and run a file in the local path that won't exist (this is the behavior I get). Changing it to ```jq``` worked fine locally after installed the package.